src/sentry/search/events/datasets/metrics.py

```diff
@@ -804,6 +805,12 @@ def _resolve_transaction_alias(self, alias: str) -> SelectType:
             alias,
         )
 
+    def _resolve_transaction_alias_on_demand(self, _: str) -> SelectType:
+        """On-demand doesn't need a transform for transaction in it's where clause
+        since conditions are saved on a per-metric basis. 
+        """
+        return Column(self.builder.resolve_column_name("transaction"))
```

Comment:
Now that I think about it after putting this up, this should probably be changed, on-demand still uses a post-normalized transaction name I believe, so we still should cover `null` and `unparameterized`, going to look tomorrow.